File: backend/MatrixMarket.Api/DTOs/UserDtos.cs
namespace MatrixMarket.Api.DTOs;

public class UserProfileResponse
{
    public int UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string StudentEmail { get; set; } = string.Empty;
    public string? Campus { get; set; }
    public int KarmaPoints { get; set; }
    public string TrustTier { get; set; } = string.Empty;
    public List<string> Badges { get; set; } = new();
    public int ListingsCount { get; set; }
    public DateTime MemberSince { get; set; }
    public int TradesCompleted { get; set; }
}

public class UpdateSettingsRequest
{
    public string? Campus { get; set; }
    public string? FullName { get; set; }
}

File: android/app/src/main/java/com/matrixmarket/app/data/remote/dto/UserDtos.kt
package com.matrixmarket.app.data.remote.dto

data class UserProfileResponse(
    val userId: Int,
    val fullName: String,
    val studentEmail: String,
    val campus: String?,
    val karmaPoints: Int,
    val trustTier: String,
    val badges: List<String>,
    val listingsCount: Int,
    val memberSince: String,
    val tradesCompleted: Int
)

data class UpdateSettingsRequest(
    val campus: String? = null,
    val fullName: String? = null
)
